lib/core/network/repository/auth_repository.dart

```diff
@@ -121,18 +139,39 @@ class AuthRepository {
       await _amplifyService.logout();
 
       /// Get firebase device token and remove to firebase
-      await _firebasePushNotificationsService.getDeviceToken().then(
-            (String? value) => value?.let((it) {
-              _firebaseDatabaseService.removeDeviceToken(it, accountName);
-            }),
-          );
+      await accountName?.let(
+        (self) async => _firebasePushNotificationsService.getDeviceToken().then(
+              (String? value) => value?.let((it) {
+                _firebaseDatabaseService.removeDeviceToken(it, self);
+              }),
+            ),
+      );
     } catch (error, stacktrace) {
       LogHelper.e('Error clearing user data', error: error, stacktrace: stacktrace);
     }
 
-    _controller.add(AuthenticationStatus.unauthenticated);
+    _controller.add(const UnAuthenticated());
     LogHelper.d('User data cleared successfully');
   }
 
   void dispose() => _controller.close();
 }
+
+sealed class AuthenticationStatus {
+  const AuthenticationStatus();
+}
+
+class Authenticated extends AuthenticationStatus {
+  final UserAuthData userAuthData;
+  final UserProfileData userProfileData;
```

Comment:
Same here - what if there is no userProfileData?

Do we create a new empty object?